crates/walrus-service/src/client/cli/runner.rs

```diff
@@ -504,6 +558,75 @@ impl ClientCommandRunner {
         ReadOutput::new(out, blob_id, blob).print_output(self.json)
     }
 
+    pub(crate) async fn read_quilt(
+        self,
+        query: QuiltPatchQuery,
+        out: Option<PathBuf>,
+        rpc_url: Option<String>,
+    ) -> Result<()> {
+        let config = self.config?;
+        let sui_read_client =
+            get_sui_read_client_from_rpc_node_or_wallet(&config, rpc_url, self.wallet).await?;
+        let read_client = Client::new_read_client_with_refresher(config, sui_read_client).await?;
+
+        let quilt_read_client = read_client.quilt_client(QuiltClientConfig::default());
+
+        let retrieved_blobs = match query.get_selector()? {
```

Comment:
Nit: We could check at the very beginning if the selector is valid, to avoid creating read clients in case of failure.